### PR TITLE
feat: Add reusable Github action for the `diagram cache`

### PR DESCRIPTION
**Pull request by Paula-Kli**
See first working usage here: https://github.com/DSD-DBS/coffee-machine/pull/8

### CODE REVIEW

File: ci-templates/github/diagram-cache/action.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+name: 'Collab Manager: Diagram Cache'
```
Comment (MoritzWeber0) [inline ci-templates/github/diagram-cache/action.yml:4]:
```suggestion
name: 'Generate diagram cache'
```

File: ci-templates/github/diagram-cache/action.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      with:
+        python-version: 3.11
+    - name: Install capellambse
+      shell: bash
+      run: pip install "capellambse[cli]@git+https://github.com/DSD-DBS/py-capellambse.git@${{ inputs.capellambse_revision }}"
+    - name: Generate diagram cache
+      shell: bash
+      run: >-
+        python -m capellambse.diagram_cache
+        --model ${{ inputs.entry_point }}
```
Comment (MoritzWeber0) [inline ci-templates/github/diagram-cache/action.yml:47]:
```suggestion
        --model "${{ inputs.entry_point }}"
```

File: ci-templates/github/diagram-cache/action.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        --output ./diagram_cache
+        --index
+        --docker ghcr.io/dsd-dbs/capella-dockerimages/capella/base:{VERSION}-${{ inputs.capella_docker_images_dropins }}-dropins-${{ inputs.capella_docker_images_revision }}
```
Comment (MoritzWeber0) [inline ci-templates/github/diagram-cache/action.yml:50]:
```suggestion
        --docker "ghcr.io/dsd-dbs/capella-dockerimages/capella/base:{VERSION}-${{ inputs.capella_docker_images_dropins }}-dropins-${{ inputs.capella_docker_images_revision }}"
```

File: ci-templates/github/diagram-cache/action.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    - name: Push diagram cache to orphan branch
+      shell: bash
+      run: |
+        DERIVED_BRANCH_NAME="$(echo diagram-cache/${{ inputs.commit_branch }})"
```
Comment (MoritzWeber0) [inline ci-templates/github/diagram-cache/action.yml:54]:
```suggestion
        DERIVED_BRANCH_NAME="diagram-cache/${{ inputs.commit_branch }}"
```

File: ci-templates/github/diagram-cache/action.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        git switch "$DERIVED_BRANCH_NAME" || git switch --orphan "$DERIVED_BRANCH_NAME"
```
Comment (MoritzWeber0) [inline ci-templates/github/diagram-cache/action.yml:55]:
```suggestion
        git switch --orphan "$DERIVED_BRANCH_NAME"
```

There is no need to preserve the derived branch, I'd say. Also, this seems to always fail with `fatal: invalid reference: diagram-cache/...`.

File: ci-templates/github/diagram-cache/action.yml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        git reset
```
Comment (MoritzWeber0) [inline ci-templates/github/diagram-cache/action.yml:56]:
```suggestion
```

The orphan branch has no history.

File: ci-templates/github/diagram-cache/action.yml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        git add diagram_cache
+        if git diff --cached --exit-code &> /dev/null || [[ "$PUSH_DIAGRAM_CACHE" == "false" ]]; then exit 0; fi
+        git config --local user.email "github-actions[bot]@users.noreply.github.com"
+        git config --local user.name "capellambse[bot]"
+        git commit -m "${{ inputs.commit_msg }}"
```
Comment (MoritzWeber0) [inline ci-templates/github/diagram-cache/action.yml:61]: We should also put the deployed commit hash in the commit information (similar to mkdocs in gh-pages).

File: ci-templates/github/diagram-cache/action.yml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        --index
+        --docker ghcr.io/dsd-dbs/capella-dockerimages/capella/base:{VERSION}-${{ inputs.capella_docker_images_dropins }}-dropins-${{ inputs.capella_docker_images_revision }}
+    - name: Push diagram cache to orphan branch
+      shell: bash
+      run: |
+        DERIVED_BRANCH_NAME="$(echo diagram-cache/${{ inputs.commit_branch }})"
+        git switch "$DERIVED_BRANCH_NAME" || git switch --orphan "$DERIVED_BRANCH_NAME"
+        git reset
+        git add diagram_cache
+        if git diff --cached --exit-code &> /dev/null || [[ "$PUSH_DIAGRAM_CACHE" == "false" ]]; then exit 0; fi
```
Comment (MoritzWeber0) [inline ci-templates/github/diagram-cache/action.yml:58]:
```suggestion
        if [[ "$PUSH_DIAGRAM_CACHE" == "false" ]]; then exit 0; fi
```

As it's the first commit on the orphan branch, there is no history and always a diff.

**Review by MoritzWeber0 — CHANGES_REQUESTED** [review]

File: ci-templates/github/diagram-cache/action.yml (hunk 2 above)
Comment (Paula-Kli) [reply ci-templates/github/diagram-cache/action.yml:47]: at the moment the entry point contains the `"` - should we change that?

File: ci-templates/github/diagram-cache/action.yml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  capella_docker_images_dropins:
+    description: 'Please have a look at https://dsd-dbs.github.io/capella-dockerimages/capella/introduction/#supported-dropins'
+    required: false
+    default: selected
+  push_diagram_cache:
+    description: 'Whether to push the diagram cache changes or not'
+    required: false
+    default: "true"
+  commit_branch:
+    description: 'The base branch to which the diagram cache belongs'
```
Comment (MoritzWeber0) [inline ci-templates/github/diagram-cache/action.yml:31]: I'd remove this input and don't make it configurable.

**Review by MoritzWeber0 — APPROVED** [review]
LGTM, just one improvement suggestion

File: ci-templates/github/diagram-cache/action.yml (hunk 9 above)
Comment (Paula-Kli) [reply ci-templates/github/diagram-cache/action.yml]: done

**Pull request merged**